gravitee-apim-gateway/gravitee-apim-gateway-handlers/gravitee-apim-gateway-handlers-api/src/main/java/io/gravitee/gateway/handlers/api/security/ApiKeyPlanBasedAuthenticationHandler.java

```diff
@@ -40,11 +40,19 @@ public ApiKeyPlanBasedAuthenticationHandler(AuthenticationHandler handler, Plan
      * {@inheritDoc}
      */
     @Override
-    protected boolean canHandleSubscription(AuthenticationContext authenticationContext) {
+    protected boolean preCheckSubscription(AuthenticationContext authenticationContext) {
```

Comment:
I don't understand the naming as there is not check or post check Subscription

Reply:
The naming is definitely not the best. It is just something with discussed with other team members and appears to be clearer than `canHandleSubscription`.

The preCheckSubscription is used in a particular case (jwt) and can be skipped under certain circumstances.